packages/components/src/toggle-group-control/toggle-group-control/as-button-group.tsx

```diff
@@ -48,7 +48,9 @@ function UnforwardedToggleGroupControlAsButtonGroup(
 	const { value, defaultValue } =
 		useComputeControlledOrUncontrolledValue( valueProp );
 
-	const [ selectedValue, setSelectedValue ] = useControlledValue( {
+	const [ selectedValue, setSelectedValue ] = useControlledValue<
+		typeof value
```

Comment:
We removed `as typeof setState` from the `useControlledValue` hook, which caused a TS error. This was resolved by explicitly specifying the generic type for `useControlledValue`.

<img width="676" height="352" alt="Screenshot 2025-07-16 at 13 31 12" src="https://github.com/user-attachments/assets/b4651e10-ee3d-4b75-a9a2-001f8052cbbc" />


https://github.com/WordPress/gutenberg/pull/70655/files#diff-4df4aa449ece25eb703b298c9bfc3cdaae55552ccd2040f629efc4caa1bf9f8fL43